README.md

```diff
@@ -1,36 +1,115 @@
-# YOURPROJECTNAME
+# Hierarchy
 
-Describe what this project is all about.
+[![Build Status](https://travis-ci.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter.svg?branch=master)](https://travis-ci.com/KohlsTechnology/hierarcbhy)
+[![Go Report Card](https://goreportcard.com/badge/github.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter)](https://goreportcard.com/report/github.com/KohlsTechnology/hierarcbhy)
+
+Hierarchy is a simple utility to merge a set of yaml or json files, based on a defined hierarchy. It is inspired by Hiera from Puppet.
 
 ## Installation
 
-Add steps for installing here.
+You can either download the binary file from the GitHub releases or [compile from source](#compiling-from-source).
 
 ## Documentation
 
-Add documentaiton here, or add a link to
-the full documentation.
+The main goal of `hierarchy` is to allow the deduplication of configuration data and allow for a fine grained control over your GitOps process. GitOps tools, like [Eunomia](https://github.com/KohlsTechnology/eunomia), can use it to generate a `values.yml` file that is then being used by [Helm](https://helm.sh) or similar tools.
+
+### Usage
+
+```
+./hierarchy -h
+usage: hierarchy [<flags>]
+
+Hierarchy
+
+Flags:
+  -h, --help                    Show context-sensitive help (also try --help-long and --help-man).
+  -f, --file="./hierarchy.lst"  Path and name of the hierarchy file.
+  -o, --output="./output.yaml"  Path and name of the output file.
+  -i, --filter="(.yaml|.yml|.json)$"
+                                Regex for file extension of files being merged
+      --trace                   Prints a diff after processing each file. This generates A LOT of output
+  -m, --failmissing             Fail if a directory in the hierarchy is missing
+  -V, --version                 Print version and build information, then exit
+  -d, --debug                   Print debug output
+
+```
+
+### Merging
+
+Hierarchy does a deep merge on the yaml structure, with the exception of lists. Lists will be completely overwritten, so choose wisely where to use them.
+
+### Hierarchy
+
+The hierarchy is defined in the file `hierarchy.lst`. This is a simple text file that lists one include folder per line and supports comments prefixed with `#`. The directories listed can be relative or absolute (try to avoid) paths. You can have directories included that are higher or lower in the structure. You have complete control. You can look at examples [here](https://github.com/KohlsTechnology/hierarchy/blob/master/testdata/).
```

Comment:
```suggestion
The hierarchy is defined in the file `hierarchy.lst`. This is a simple text file that lists one include folder per line and supports comments prefixed with `#`. The directories listed can be relative or absolute (try to avoid) paths. You can have directories included that are higher or lower in the structure to control their precedence. You can look at examples [here](https://github.com/KohlsTechnology/hierarchy/blob/master/testdata/).
```